yb-voyager/cmd/assessMigrationCommandBulk.go

```diff
@@ -0,0 +1,311 @@
+		if err != nil {
+			log.Errorf("failed to assess migration for schema %s: %v", dbConfig.Schema, err)
+			fmt.Printf("failed to assess migration for schema %s: %v\n", dbConfig.Schema, err)
+			if !continueOnError {
+				break
+			}
+		}
+	}
+
+	err = generateBulkAssessmentReport(dbConfigs)
+	if err != nil {
+		utils.ErrExit("failed to generate bulk assessment report: %s", err)
+	}
+}
+
+func executeAssessment(dbConfig AssessMigrationDBConfig) error {
+	log.Infof("executing assessment for schema %q", dbConfig.Schema)
+	exportDirPath := getAssessmentExportDirPath(dbConfig)
+	cmdArgs := buildCommandArguments(dbConfig, exportDirPath)
+	if err := os.MkdirAll(exportDirPath, 0755); err != nil {
+		return fmt.Errorf("creating export-directory %q for schema %q: %w", exportDirPath, dbConfig.Schema, err)
+	}
+
+	execCmd := exec.Command(os.Args[0], cmdArgs...)
+	execCmd.Stdout = os.Stdout
+	execCmd.Stderr = os.Stderr
+	execCmd.Stdin = os.Stdin
+	log.Infof("executing the cmd: %s", execCmd.String())
+	err := execCmd.Run()
+	if err != nil {
+		return fmt.Errorf("error while assess migration of schema-%s: %v", dbConfig.Schema, err)
+	}
+	return nil
+}
+
+func buildCommandArguments(dbConfig AssessMigrationDBConfig, exportDirPath string) []string {
+	log.Infof("building assess-migration command arguments for schema %q", dbConfig.Schema)
+	args := []string{"assess-migration",
+		"--source-db-type", dbConfig.DbType,
+		"--source-db-schema", dbConfig.Schema,
+		"--export-dir", exportDirPath,
+	}
+
+	if dbConfig.User != "" {
+		args = append(args, "--source-db-user", dbConfig.User)
+	}
+	if dbConfig.Password != "" {
+		// password either has to be provided via fleet_config_file or can be provided at run-time.
+		// env var route is not supported for assess-migration-bulk command
+		args = append(args, "--source-db-password", dbConfig.Password)
+	}
+	if dbConfig.TnsAlias != "" {
+		args = append(args, "--oracle-tns-alias", dbConfig.TnsAlias)
+	}
+	if dbConfig.SID != "" {
+		args = append(args, "--source-db-name", dbConfig.SID)
+	}
+	if dbConfig.Host != "" {
+		args = append(args, "--source-db-host", dbConfig.Host)
+	}
+	if dbConfig.Port != "" {
+		args = append(args, "--source-db-port", dbConfig.Port)
+	}
+	return args
+}
+
+func parseFleetConfigFile(filePath string) ([]AssessMigrationDBConfig, error) {
+	log.Infof("parsing fleet config file %q", filePath)
+	var dbConfigs []AssessMigrationDBConfig
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+		dbConfig := parseFleetConfigLine(line)
+		dbConfigs = append(dbConfigs, dbConfig)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return dbConfigs, nil
+}
+
+/*
+Format: covers both SSL and non-SSL cases
+
+	<dbtype>,<hostname>,<port>,<service_name>,<sid>,<tns_alias>,<username>,<password>,<schema>
```

Comment:
Ques: With this structure, the user will have to specify all but in the case of non-SSL, SSL params will be empty?

Reply:
yes right. 
What i am think to add here is a csv header so that:
- file is more readable
- flexiblity for user to define the parameters here

Reply:
Header makes sense I think! So, in that case, it is possible for a user to only specify a susbet of all the possible arguments as the names of the arguments will be in header. Correct?

Reply:
yeah specifying a subset, and ignoring the not required ones will be possible with the header.

Reply:
https://yugabyte.atlassian.net/browse/DB-12385